elastic/kibana
# [Grok] Migrate code editors to monaco

Partially addresses: https://github.com/elastic/kibana/issues/107215

## Summary

This PR removes the dependency it has on ace editor in favour of Monaco. Note that `CodeEditor` is a kibana wide abstraction built on top of Monaco and located at `@kbn/kibana-react-plugin/public`.

**_Note:_**: The new Monaco based editor doesnt have support for grok syntax highlighting, but I've created an issue for the sharedux team to help us out with that https://github.com/elastic/kibana/issues/155506


#### Migrated to `EuiCodeBlock`
- `components/event_output/event_output.js`

#### Migrated to `Monaco`
- `components/custom_patterns_input/custom_patterns_input.js`
- `components/event_input/event_input.js`
- `components/pattern_input/pattern_input.js`

**How to test**
1. Start elasticsearch with `yarn es snapshot --license=trial -E path.repo=/tmp/es-backups` and kibana with `yarn start`
2. Navigate to `Stack Management` -> `DevTools` -> `Grok`
3. Verify that all editors are Monaco

<details>
<summary>Screenshots</summary>

![Screenshot 2023-04-21 at 11 56 18](https://user-images.githubusercontent.com/1191206/233607396-4f064b4f-334f-4a30-a43a-9b9675fb8bba.png)


</details>
